### DISCUSSION TITLE
Title: Request to Modify sortComparer to Access Full Row Data

### DISCUSSION THREAD

**Discussion by hwlv** [ghiscoding/slickgrid-universal#1712] [Ideas]
Description:

Hi,

I am currently using SlickGrid Universal and would like to request an enhancement to the sortComparer function. At present, the sortComparer function only receives the values of the current column (value1, value2) for comparison. However, in some cases, it's more flexible and useful to access the full row's data to implement custom sorting logic that takes other columns into account.
Proposed Solution:

Modify the sortComparer signature or provide an additional parameter that includes the entire row data for value1 and value2.
This would allow sorting not only based on the current column's value but also on other column values or custom logic that requires full row context.

For example:
```js
sortComparer: (value1: any, value2: any, sortDirection?: SortDirectionNumber, sortColumn?: Column, gridOptions?: GridOption, rowData1?: any, rowData2?: any) => {
  // Access to full row data (rowData1, rowData2) in addition to current column values
  const compare = rowData1.someOtherField - rowData2.someOtherField;
  return sortDirection * compare;
}
```

This would provide more flexibility for custom sorting.

Thank you for considering this request! I'm happy to provide further details if needed.

**Comment by ghiscoding** [comment]
if you want to sort by another column, simply use `queryFieldSorter`

https://github.com/ghiscoding/slickgrid-universal/blob/26fa5aacfe73d326a4a9b4f79043e788007e28bf/packages/common/src/interfaces/column.interface.ts#L294-L298
